vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp

```diff
@@ -1184,15 +1199,31 @@ void WindowsTextInputComponentView::OnTextUpdated() noexcept {
   data.mostRecentEventCount = m_nativeEventCount;
 
   m_state->updateState(std::move(data));
+  std::string richEditText = GetTextFromRichEdit();
 
+  // call onChange
   if (m_eventEmitter && !m_comingFromJS) {
     auto emitter = std::static_pointer_cast<const facebook::react::WindowsTextInputEventEmitter>(m_eventEmitter);
     facebook::react::WindowsTextInputEventEmitter::OnChange onChangeArgs;
-    onChangeArgs.text = GetTextFromRichEdit();
+    onChangeArgs.text = richEditText;
     onChangeArgs.eventCount = ++m_nativeEventCount;
     emitter->onChange(onChangeArgs);
   }
 
+  // call onKeyPress
+  if (m_eventEmitter && !m_comingFromJS && !richEditText.empty()) {
```

Comment:
Why only fire the event if there's text? Is it expected that pressing, say, backspace in an empty textbox that new KeyPress event gets fired?

Reply:
oops, you are right! fixed it so it does fire with backspace in a empty textbox :)